agent/php_user_instrument.c

```diff
@@ -515,15 +431,28 @@ void nr_php_reset_user_instrumentation(void) {
   nruserfn_t* p = nr_wrapped_user_functions;
 
   while (0 != p) {
+#if ZEND_MODULE_API_NO >= ZEND_7_4_X_API_NO
+    p->zf = NULL;
+#endif
     p->is_wrapped = 0;
     p = p->next;
   }
+#if ZEND_MODULE_API_NO >= ZEND_7_4_X_API_NO
+  if (NULL != user_function_wrappers) {
+    nr_hashmap_destroy(&user_function_wrappers);
+  }
+  /* wraprecs are re-usable and stored in linked list - no destructor */
+  user_function_wrappers = nr_hashmap_create(NULL);
+#endif
 }
 
 /*
  * Remove any transient wraprecs. This must only be called on request shutdown!
  */
 void nr_php_remove_transient_user_instrumentation(void) {
+#if ZEND_MODULE_API_NO >= ZEND_7_4_X_API_NO
```

Comment:
Create a supportability metric with number of wraprecs for transient user functions. What should be the name of the metric?

Reply:
This will be added in a separate PR.